dependencies/adi/command_parser/command_parser.cpp

```diff
@@ -0,0 +1,53 @@
+/**********************************************************************************/
+/* MIT License                                                                    */
+/*                                                                                */
+/* Copyright (c) 2021 Analog Devices, Inc.                                        */
+/*                                                                                */
+/* Permission is hereby granted, free of charge, to any person obtaining a copy   */
+/* of this software and associated documentation files (the "Software"), to deal  */
+/* in the Software without restriction, including without limitation the rights   */
+/* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      */
+/* copies of the Software, and to permit persons to whom the Software is          */
+/* furnished to do so, subject to the following conditions:                       */
+/*                                                                                */
+/* The above copyright notice and this permission notice shall be included in all */
+/* copies or substantial portions of the Software.                                */
+/*                                                                                */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     */
+/* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       */
+/* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    */
+/* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         */
+/* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  */
+/* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  */
+/* SOFTWARE.                                                                      */
+/**********************************************************************************/
+
+#include "command_parser.h"
+#include <iostream>
+#include <string>
+#include <vector>
+
+std::vector<std::pair<std::string, std::string>>
+CommandParser::getConfiguration() {
+    return m_command_vector;
+}
+
+int CommandParser::parseArguments(int argc, char *argv[]) {
+    for (int i = 1; i < argc; i++) {
+        int end = std::string(argv[i]).find("=");
+        int config = std::string(argv[i]).find(".json");
```

Comment:
This implementation assumes that an argument is a string having the suffix ".json". This is not ok because is not generic. What happens at some point a client app wants to use a .cfg instead of .json?
Please make this more generic.

Reply:
Fixed, now also works with .cfg

Reply:
That's not generic.
What happens if at some point a client app wants to use .ini instead of .json?
What happens if at some point a client app wants to use .png or .jpeg? Or any other possible extension? (There are thousands of possibilities).
What happens if at some point a client app wants to use the name of a file that has no extension?

Reply:
Changed parser to be generic